Explorer/Assets/DCL/NftPrompt/NftPromptController.cs

```diff
@@ -0,0 +1,198 @@
+        {
+            if (viewInstance.SpinnerNftImage == null)
+                return;
+
+            viewInstance.SpinnerNftImage.SetActive(isVisible);
+        }
+
+        private void ShowMainLoading(bool isVisible)
+        {
+            if (viewInstance.SpinnerGeneral == null)
+                return;
+
+            viewInstance.SpinnerGeneral.SetActive(isVisible);
+        }
+
+        private void ShowMainErrorFeedback(bool isVisible)
+        {
+            if (viewInstance.MainErrorFeedbackContent == null)
+                return;
+
+            if (isVisible)
+                ShowMainLoading(false);
+
+            viewInstance.MainErrorFeedbackContent.SetActive(isVisible);
+        }
+
+        private void SetNftInfo(NftInfo info)
+        {
+            ShowMainLoading(false);
+            ShowMainErrorFeedback(false);
+            viewInstance.NftContent.SetActive(true);
+            viewInstance.TextNftName.text = info.name;
+
+            bool hasMultipleOwners = info.owners.Length > 1;
+            if (hasMultipleOwners)
+                viewInstance.TextMultipleOwner.text = string.Format(MULTIPLE_OWNERS_FORMAT, info.owners.Length);
+            else
+            {
+                viewInstance.TextOwner.text = info.owners.Length == 1
+                    ? FormatOwnerAddress(info.owners[0].address, ADDRESS_MAX_CHARS)
+                    : OWNER_NOT_AVAILABLE;
+            }
+
+            viewInstance.TextOwner.gameObject.SetActive(!hasMultipleOwners);
+            viewInstance.MultipleOwnersContainer.gameObject.SetActive(hasMultipleOwners);
+
+            bool hasDescription = !string.IsNullOrEmpty(info.description);
+            if (hasDescription)
+                viewInstance.TextDescription.text = info.description;
+
+            viewInstance.ContainerDescription.SetActive(hasDescription);
+
+            viewInstance.TextOpenMarketButton.text = "VIEW";
+            if (!string.IsNullOrEmpty(info.marketName))
+                viewInstance.TextOpenMarketButton.text = $"{viewInstance.TextOpenMarketButton.text} ON {info.marketName.ToUpper()}";
+
+            marketUrl = null;
+            if (!string.IsNullOrEmpty(info.marketLink))
+                marketUrl = info.marketLink;
+            else if (!string.IsNullOrEmpty(info.assetLink))
+                marketUrl = info.assetLink;
+
+            placeImageController.SetImage(null);
+            placeImageController.RequestImage(info.imageUrl);
+
+            lastNftInfo = info;
+        }
+
+        private string FormatOwnerAddress(string address, int maxCharacters)
+        {
+            const string ELLIPSIS = "...";
+
+            if (address.Length <= maxCharacters)
+                return address;
+
+            int segmentLength = Mathf.FloorToInt((maxCharacters - ELLIPSIS.Length) * 0.5f);
+            return $"{address[..segmentLength]}{ELLIPSIS}{address.Substring(address.Length - segmentLength, segmentLength)}";
```

Comment:
If you look, in this case, we're not putting the ellipsis at the end of the string (in that case we could of course use the TextMeshPro's capabilities) but we're putting it in the middle of the string because, by design, we want to have the address in this way:

![image](https://github.com/decentraland/unity-explorer/assets/64659061/832aa16a-83f9-4daa-9dcd-d07d88f04e32)

This is the reason why we do it from code.

Reply:
you're right!